airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js

```diff
@@ -284,11 +286,11 @@ define( [ "util/lang", "util/uri", "util/xhr", "util/keys", "util/mediatypes", "
     resetInput();
   }
 
-  function onSuccess( data ) {
+  function onSuccess( data, imported ) {
     var el = _GALLERYITEM.cloneNode( true ),
         source = data.source;
 
-    if ( !_media.clipData[ source ] ) {
+    if ( !_media.clipData[ source ]  || imported !== undefined) {
```

Comment:
Don't forget that comment. I think it's an important underlying design thing to understand.